### ISSUE TITLE
Strange behavior with Project References on

### ISSUE THREAD

**Issue by kirill-konshin** [TypeStrong/ts-loader#997]
I have a project where I'm using `ts-loader@6.1.0` with `projectReferences: true` and following `tsconfig`:

```json
{
  "extends": "xxx/typescript/tsconfig.json",
  "compilerOptions": {
    "composite": true,
    "incremental": true,
    "tsBuildInfoFile": "./.tscache/tsconfig.tsbuildinfo",
    "target": "es5",
    "module": "commonjs",
    "baseUrl": ".",
    "rootDir": "./src",
    "outDir": "./lib",
    "declaration": true,
    "declarationDir": "./lib",
    "typeRoots": [
      "./node_modules/@types"
    ],
    "paths": {
      "web-modules-core": [
        "../core"
      ],
    }
  },
  "include": [
    "src/**/*.ts",
    "src/**/*.tsx"
  ],
  "references": [
    {
      "path": "../core"
    },
  ]
}
```

The project is a lerna monorepository.

On first build I get error that `web-modules-core` can't be resolved. Even though `web-modules-core` package (located in `../core`) builds itself during the startup.

If I stop webpack and start it again it succeeds...

**Comment by johnnyreilly** [comment]
Could you provide a minimal reproduction repo please? 

cc @andrewbranch

**Comment by kirill-konshin** [comment]
@johnnyreilly @andrewbranch I can create a private repo and add you there if this would work.

**Comment by kirill-konshin** [comment]
In fact, I was able to make it clean enough to be public: https://github.com/kirill-konshin/wmc-ts-bug

```bash
$ npm install
$ cd storybook
$ npm start
```

Which consistently results in

```
ERROR in /Users/dis/Sites/wmc-ts-bug/storybook/src/stories/Well.tsx
./src/stories/Well.tsx
[tsl] ERROR in /Users/dis/Sites/wmc-ts-bug/storybook/src/stories/Well.tsx(4,20)
      TS2307: Cannot find module 'web-modules-core'.
```

Then `Ctrl+C`

Then again

```bash
$ npm start
```

This time it will work just fine.

In order to clean everything up use `npx lerna run clean` which will remove all build artifacts.

**Comment by johnnyreilly** [comment]
Thanks @kirill-konshin!

**Comment by kirill-konshin** [comment]
Any luck so far @johnnyreilly ?

**Comment by johnnyreilly** [comment]
I'm not looking at this myself I'm afraid - completely snowed. It's possible it may be resolved by @sheetalkamat's work on https://github.com/TypeStrong/ts-loader/pull/1003

**Comment by Bessonov** [comment]
Activity!

**Comment by johnnyreilly** [comment]
Do you know anything about this @andrewbranch? I haven't looked at this at all I'm afraid..

**Comment by jleider** [comment]
I filed a similar bug a while ago: https://github.com/TypeStrong/ts-loader/issues/1005 and I'm just now getting back to trying to implement project references again. As you state the first build fails but still produces the referenced projects definition files. Thats why the second build succeeds because those definition files are available at the onset of the compilation the second time around. It seems like a race condition or caching issue on ts-loader's part since the project references work just fine using `tsc --build`

**Comment by Bessonov** [comment]
For the community the stale bots and friends are the worst thing ever :/

**Comment by Bessonov** [comment]
Let's notify everyone just for fun and stale bot. Do you like it, do you? I don't know why @johnnyreilly think that stale bot is a good idea.

**Comment by andrewbranch** [comment]
I’ve pinned it to keep the bot from marking it as stale again.

Keep in mind that ts-loader is maintained in the spare time of folks who have full-time jobs, and for many years, that has _overwhelmingly_ been @johnnyreilly. I can’t speak for him, but I don’t currently don’t have the time or energy to devote to this issue. I understand your frustration, but, I’d wager that any tool that lets @johnnyreilly volunteer his time to this project without it taking over his life and becoming unmanageable is a net win for the community.

Expressing your frustration is ok, and it’s useful to know people still care about this issue. I’d just also remind you that a) it’s not anybody’s job to do this, and b) it _can_ be _anybody’s_ job to do this—it’s open source, and if you open a PR, I would make time to review it 🙂

**Comment by appzuka** [comment]
This seems to have been fixed by PR [#1136](https://github.com/TypeStrong/ts-loader/pull/1136) by @sheetalkamat.  This was to fix [#1114](https://github.com/TypeStrong/ts-loader/issues/1114) which had the same symptoms you reported.  Your repo does not seem to have the symlinks present in [#1114](https://github.com/TypeStrong/ts-loader/issues/1114), which were reported to be causing the issue.  However, the changes made to fix that issue may have also fixed yours.

I cloned your wmc-ts-bug repo, upgraded ts-loader to 8.0.0 and typescript to 3.9.3.  It then builds and runs without error.

**Comment by andrewbranch** [comment]
Fantastic! Thanks for investigating @appzuka! Sounds like we can close this then.
